Makefile

```diff
@@ -99,6 +107,44 @@ crystal: $(O)/crystal ## Build the compiler
 deps: $(DEPS) ## Build dependencies
 llvm_ext: $(LLVM_EXT_OBJ)
 
+.PHONY: install
+install: $(O)/crystal man/crystal.1.gz ## Install the compiler at DESTDIR
+	$(INSTALL) -D -m 0755 "$(O)/crystal" "$(BINDIR)/crystal"
+
+	$(INSTALL) -d -m 0755 $(DATADIR)
+	cp -av src "$(DATADIR)/src"
```

Comment:
At least Ruby and Python put their standard library into `/usr/lib`. Why is `/usr/share` better? What's the advantage of including `src` subdirectory?

Reply:
Again, this structure is directly copied from the tarball generation in distribution-scripts. 

I don't know the specific reasons for lib vs. share. But in the Filesystem Hierarchy Standard, ˋ/usr/libˋ is for (binary) libraries and ˋ/usr/shareˋ for architecture-independent data. Ths stdlib source code is architecture-independent, so I suppose it makes sense to place it there. Not sure why Ruby and Python put it in ˋlibˋ.

The reason for the ˋsrcˋ folder is because there's other data that is to be placed beside it, for example ˋdocsˋ and ˋsamplesˋ.

Reply:
> /usr/lib includes object files and libraries.  On some systems, it may also include internal binaries that are not intended to be executed directly by users or shell scripts.
>
> Applications may use a single subdirectory under /usr/lib. If an application uses a subdirectory, all architecture-dependent data exclusively used by the application must be placed within that subdirectory. 

https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s06.html



Where does this say architecture dependent or binary?

Just looking randomly through my `/usr/lib` I find "text" files by:

* Go
* NodeJS
* Ruby
* Python
* TCL

* udev
* systemd
* resolveconf

Furthermore precompiled libraries by 

* Erlang
* OCaml

`/usr/share` seems preferred by Perl and Vala only from a quick look.

Reply:
I agree, the exact semantics of the specification may be subject to interpretation. They're not completely clear.

But I don't think this really matters much. `/usr/share/crystal` is not *wrong*, so there's no reason against it. We're using this path in all distribution packages (at least for linux, not sure about mac). I don't see much reason to change it.

Reply:
No need for that weird `src` subdirectory? :P